lgc/builder/DescBuilder.cpp

```diff
@@ -500,12 +500,13 @@ Value *DescBuilder::scalarizeIfUniform(Value *value, bool isNonUniform) {
 //
 // @param bufferDesc : The buffer descriptor to query.
 // @param instName : Name to give instruction(s).
-Value *DescBuilder::CreateGetBufferDescLength(Value *const bufferDesc, const Twine &instName) {
+Value *DescBuilder::CreateGetBufferDescLength(Value *const bufferDesc, unsigned offset, const Twine &instName) {
   // In future this should become a full LLVM intrinsic, but for now we patch in a late intrinsic that is cleaned up
   // in patch buffer op.
   std::string callName = lgcName::LateBufferLength;
-  addTypeMangling(nullptr, bufferDesc, callName);
-  return CreateNamedCall(callName, getInt32Ty(), bufferDesc, Attribute::ReadNone);
+  Value *args[] = {bufferDesc, getInt32(offset)};
+  addTypeMangling(nullptr, args, callName);
```

Comment:
We don't really need mangling here, as the args always have the same types. I realize you didn't add the mangling. :-)

Reply:
Fixed. offset is used as Value*

Reply:
I think you deleted the wrong line. :-)

Reply:
Oops, no, I misread the github UI. Sorry.

Reply:
But we could still remove the addTypeMangling, and just use lgcName::LateBufferLength directly.